hyphen.c

```diff
@@ -1076,17 +1076,20 @@ int hnj_hyphen_norm(const char *word, int word_size, char * hyphens,
 void hnj_hyphen_hyphword(const char * word, int l, const char * hyphens, 
     char * hyphword, char *** rep, int ** pos, int ** cut)
 {
-  int hyphenslen = l + 5;
+  /* hyphword buffer size must be at least 2 * l */
+  int hyphword_len = 2 * l - 1;
```

Comment:
Since we're starting to be precise, should this also check for possible overflow?

Just in case, these were helpful for me:

http://stackoverflow.com/questions/1815367/multiplication-of-large-numbers-how-to-catch-overflow

https://www.securecoding.cert.org/confluence/display/c/INT32-C.+Ensure+that+operations+on+signed+integers+do+not+result+in+overflow